pennylane/templates/subroutines/approx_time_evolution.py

```diff
@@ -0,0 +1,156 @@
+
+    for some Hamiltonian of the form:
+
+    .. math:: H \ = \ \displaystyle\sum_{n} H_n.
+
+    In general, implementing this unitary with a set of quantum gates is difficult, as the terms :math:`H_n` don't
+    necessarily commute with one another. However, we are able to exploit the Trotter-Suzuki decomposition formula:
+
+    .. math:: e^{A \ + \ B} \ = \ \lim_{n \to \infty} \Big[ e^{A/n} e^{B/n} \Big]^n
+
+    to implement an approximation of the time-evolution operator as:
+
+    .. math:: U \ \approx \ \displaystyle\prod_{k \ = \ 1}^{n} \displaystyle\prod_{n} e^{-i H_n t / N},
+
+    with the approximation becoming better for larger :math:`n`. It is also important to note that
+    this decomposition is exact for any value of :math:`n` when each term of the Hamiltonian, :math:`H_n`,
+    commutes with every other term.
+
+    .. note::
+
+       This template uses the ``PauliRot`` operation in order to implement
+       exponentiated terms of the input Hamiltonian. This operation only takes
+       terms that are explicitly written in terms of products of Pauli matrices (``PauliX``, ``PauliY``, ``PauliZ``, and ``Identity``).
+       Thus, each term in the Hamiltonian must be expressed this way upon input, or else an error will be raised.
+
+    Args:
+        hamiltonian (pennylane.Hamiltonian): The Hamiltonian defining the
+           time-evolution operator. The indices of the observables in the Hamiltonian correspond to the index of
+           the ``wires`` element to which each observable is being applied.
+           The Hamiltonian must be explicitly written
+           in terms of products of Pauli gates (X, Y, Z, and I).
+
+        wires (Iterable or Wires): The wires on which the template is applied.
+
+        time (int or float): The time of evolution, namely the parameter ;math:`t` in :math:`e^{- i H t}`.
+
+        n (int): The number of Trotter steps used when approximating the time-evolution operator.
+
+    Raises:
+        ValueError: if inputs do not have the correct format
+
+    .. UsageDetails::
+
+        The template is used inside a qnode:
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import ApproxTimeEvolution
+
+            n_wires = 2
+            wires = range(n_wires)
+
+            dev = qml.device('default.qubit', wires=n_wires)
+
+            coeffs = [1, 1]
+            obs = [qml.PauliX(0), qml.PauliX(1)]
+            hamiltonian = qml.Hamiltonian(coeffs, obs)
+
+            @qml.qnode(dev)
+            def circuit(time):
+                TimeEvolution(hamiltonian, wires, time, n=1)
+                return [qml.expval(qml.PauliZ(wires=i)) for i in wires]
+
+        >>> circuit(1)
+        [-0.41614684 -0.41614684]
+    """
+
+    pauli = {"Identity": "I", "PauliX": "X", "PauliY": "Y", "PauliZ": "Z"}
+
+    ###############
+    # Input checks
+
+    wires = Wires(wires)
+
+    if not isinstance(hamiltonian, qml.vqe.vqe.Hamiltonian):
```

Comment:
it's strange to have to use `vqe.vqe`. Something to keep in mind as a potential refactor